docs/spec/concepts.rst

```diff
@@ -3,6 +3,336 @@
+**assignable to** over "consistent subtype of". The two are synonymous, but
+"assignable to" is shorter, and may communicate a clearer intuition to many
+readers.
+
+For example, ``Any`` is :term:`assignable` to ``int``, because ``int`` is a
+materialization of ``Any``, and ``int`` is a subtype of ``int``. The same
+materialization also shows that ``int`` is assignable to ``Any``.
+
+The assignable-to relation is not generally symmetric, however. If ``B`` is a
+subtype of ``A``, then ``tuple[Any, B]`` is assignable to ``tuple[int, A]``,
+because ``tuple[Any, B]`` can materialize to ``tuple[int, B]``, which is a
+subtype of ``tuple[int, A]``. But ``tuple[int, A]`` is not assignable to
+``tuple[Any, B]``.
+
+For a gradual structural type, consistency and assignability are also
+structural. For example, the structural type "all objects with an attribute
+``x`` of type ``Any``" is consistent with (and assignable to) the structural
+type "all objects with an attribute ``x`` of type ``int``".
+
+Summary of type relations
+-------------------------
+
+The subtype, supertype, and equivalence relations establish a partial order on
+fully static types. The analogous relations on gradual types (via
+materialization) are "assignable-to" (or "consistent subtype"),
+"assignable-from" (or "consistent supertype"), and "consistent with". We can
+visualize this analogy in the following table:
+
+.. list-table::
+   :header-rows: 1
+
+   * - Fully static types
+     - Gradual types
+   * - ``B`` is a :term:`subtype` of ``A``
+     - ``B`` is :term:`assignable` to (or a consistent subtype of) ``A``
+   * - ``A`` is a :term:`supertype` of ``B``
+     - ``A`` is assignable from (or a consistent supertype of) ``B``
+   * - ``B`` is :term:`equivalent` to ``A``
+     - ``B`` is :term:`consistent` with ``A``
+
+We can also define equivalence on gradual types. Two gradual types ``A`` and
+``B`` are equivalent (that is, the same gradual type, not merely consistent
+with one another) if and only if all materializations of ``A`` are also
+materializations of ``B``, and all materializations of ``B`` are also
+materializations of ``A``.
+
+Attributes and methods
+----------------------
+
+In Python, we can do more with objects at runtime than just assign them to
+names, pass them to functions, or return them from functions. We can also
+get/set attributes and call methods.
+
+In the Python object model, the operations that can be performed on a value all
+desugar to method calls. For example, ``a + b`` is (roughly, eliding some
+details) syntactic sugar for either ``type(a).__add__(a, b)`` or
+``type(b).__radd__(b, a)``.
+
+For a static type checker, accessing ``a.foo`` is a type error unless all
+possible objects in the set represented by the type of ``a`` have the ``foo``
+attribute. (We consider an implementation of ``__getattr__`` to be a getter for
+all attribute names, and similarly for ``__setattr__`` and ``__delattr__``.)
```

Comment:
I agree this section is slightly awkward currently, because I don't actually want to give a full specification of attribute access on types (that's deserving of its own chapter, for sure in a separate PR), but I do think that it's important to understanding the meaning of types (which is the purpose of the concepts doc) to have _some_ discussion of attribute access, not just assignability, since this is such a common operation on types.

So I definitely don't want to get into `__getattribute__` here; if anything I'd prefer to not even digress into `__getattr__`, because it does start to sound more like a specification of attribute access, which I don't want to get into here.

I think my preferred option would be to replace this parenthetical with just a note that attribute access is actually more complicated than that, and a link to the data model. I'll push that change; hopefully that's agreeable to everyone who has already done their final review.

Reply:
I ended up leaving in the mention of `__getattr__` and `__setattr__`, since it has already been reviewed and seems more useful to have it than not.

I added a link to the data model doc and an acknowledgment that there are more complexities.

In the future if we do have a full spec of attribute access, this section could perhaps be slightly shortened and link to that instead.